classes/core/PKPQueueProvider.php

```diff
@@ -145,6 +147,36 @@ public function boot()
                 ])
             );
         });
+
+        Queue::createPayloadUsing(function(string $connection, string $queue, array $payload) {
+            
+            // If a job dispatch on CLI (e.g. TestJobSuccess/TestJobFailure),
+            // there is simple no way to get the context(journal/press/server) information
+            // and no need to set the context id information in job payload
+            if (runOnCLI()) {
+                return [];
+            }
+
+            return [
+                'context_id' => Application::get()->getRequest()->getContext(),
+            ];
+        });
```

Comment:
we need the context information and as mostly will be dispatched at web request life cycle, we can have the context information . But attaching it to job data is risky as some jobs already have it and it may cause unforseen impact, just adding it to payload is safe as it that has no impact on serialisation of data . 

only catch is that is a job dispatch another job as in those cases we simple have no way to determine the context information when running on CLI .